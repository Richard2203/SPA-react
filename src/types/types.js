// centralizando los strings/opciones del authReducer.
// Se centralizan en un objeto para evitar errores de tipeo
export const types = {
	login: '[auth] login',
	logout: '[auth] logout',
};
