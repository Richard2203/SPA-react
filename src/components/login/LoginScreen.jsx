import React from 'react';

const LoginScreen = () => {
	return (
		<div>
			<h2>LoginScreen</h2>
			<hr />
		</div>
	);
};

export default LoginScreen;
