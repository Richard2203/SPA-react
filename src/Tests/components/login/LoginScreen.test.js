import { mount } from 'enzyme';
import { Router } from 'react-router-dom';
import { AuthContext } from '../../../auth/AuthContext';
import LoginScreen from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('Pruebas en el componente LoginScreen', () => {
	const historyMock = {
		push: jest.fn(),
		goBack: jest.fn(),
		location: { pathname: '/login' },
		listen: jest.fn(),
		createHref: jest.fn(),
		replace: jest.fn(),
	};

	const contextValue = {
		dispatch: jest.fn(),
		user: {
			name: 'paquito',
			logged: true,
		},
	};

	const wrapper = mount(
		<AuthContext.Provider value={contextValue}>
			<Router history={historyMock}>
				<LoginScreen />
			</Router>
		</AuthContext.Provider>
	);

	test('debe renderizar correctamente el componente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('debe de realizar el disptch y la navegacion', () => {
		wrapper.find('button').prop('onClick')();
		expect(contextValue.dispatch).toHaveBeenCalledWith({
			type: types.login,
			payload: { name: 'paquito' },
		});

		expect(historyMock.replace).toHaveBeenCalledWith('/');
	});

	// Aqui implementamos localStorage directamente mientras que en otras
	// pruebas implementamos localStorage mediante un mock, la razon es que
	// aqui no haremos pruebas referentes a cuantas veces se llama, como se llamo
	// con que argumentos, etc; simplemente ejecutamos localStorage y ya
	test('debe de ser llamado localStorage', () => {
		const handleLogin = wrapper.find('button').prop('onClick');

		// disparo handleLogin teniendo localStorage vacio por lo cual
		// redirecciona a '/'
		handleLogin();
		expect(historyMock.replace).toHaveBeenCalledWith('/');

		localStorage.setItem('lastPath', '/dc');
		// disparo handleLogin teniendo localStorage con una ruta y por ende
		// debe redireccionar a la ruta guardada
		handleLogin();
		expect(historyMock.replace).toHaveBeenCalledWith('/dc');
	});
});
