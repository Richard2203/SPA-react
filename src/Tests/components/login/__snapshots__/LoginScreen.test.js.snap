// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente LoginScreen debe renderizar correctamente el componente 1`] = `
<div>
  <h2>
    LoginScreen
  </h2>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
